test/integration/errors_test.exs

```diff
@@ -45,4 +48,49 @@ defmodule ErrorsTest do
     assert_raise Error, fn -> Xandra.prepare!(conn, "") end
     assert_raise Error, fn -> Xandra.execute!(conn, "USE unknown") end
   end
+
+  describe "on Xandra.Cluster level" do
+    defmodule Xandra.Cluster.PoolMock do
+      @behaviour :gen_statem
+
+      def child_spec(opts) do
+        %{
+          id: __MODULE__,
+          start: {__MODULE__, :start_link, [opts]},
+          type: :worker,
+          restart: :permanent
+        }
+      end
+
+      def start_link([]) do
+        :gen_statem.start_link(__MODULE__, :no_args, [])
+      end
+
+      def checkout(pid) do
+        :gen_statem.call(pid, :checkout)
+      end
+
+      @impl true
+      def init(:no_args) do
+        {:ok, random_pid} = Task.start(fn -> nil end)
```

Comment:
To avoid flaky tests, let's make sure that we spawn a PID and wait for it to die.

```suggestion
        {dead_pid, ref} = spawn_monitor(fn -> :ok end)

        receive do
          {:DOWN, ^ref, _, _, _} -> :ok
        end
```